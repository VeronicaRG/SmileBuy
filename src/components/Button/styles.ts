import styled from 'styled-components/native';

export const Container = styled.TouchableOpacity`
  width: 80%;
  height: 50px;
  background-color: ${({ theme }) => theme.colors.tertiary};
  box-shadow: 0px 7px 5px rgba(194, 193, 248, 0.3);
  margin: ${({ theme }) => theme.sizes.spaces.x5}px 0;
  border-radius: ${({ theme }) => theme.sizes.radius.x10}px;
  align-self: center;
  justify-content: center;
  align-items: center;
`;
