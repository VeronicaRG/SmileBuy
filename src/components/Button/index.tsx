import React from 'react';

import { useTheme } from 'styled-components';

import BaseText from '@components/BaseText';

import { Container } from './styles';
import { ButtonProps } from './types';

const Button: React.FC<ButtonProps> = ({ title, onPress }) => {
  const { colors } = useTheme();
  return (
    <Container onPress={onPress}>
      <BaseText size="h6" color={colors.fixNeutral.white}>
        {title}
      </BaseText>
    </Container>
  );
};

export default Button;
