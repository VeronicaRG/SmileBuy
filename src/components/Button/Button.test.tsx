import React from 'react';

import { fireEvent, render } from '@testing-library/react-native';

import Theme from '../../theme';
import Button from './index';

import 'jest-styled-components';

describe('Button', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <Theme>
        <Button title="teste" />
      </Theme>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should call onPress when the button was pressed', () => {
    const fnMock = jest.fn();
    const { getByTestId } = render(
      <Theme>
        <Button title="teste" onPress={fnMock} testID={'btn_test'} />
      </Theme>,
    );

    const touchable = getByTestId('btn_test');
    fireEvent.press(touchable);

    expect(fnMock).toBeCalled();
  });
});
