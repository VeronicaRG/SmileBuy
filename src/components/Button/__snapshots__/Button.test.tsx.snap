// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#504DB6",
      "borderRadius": 40,
      "height": 50,
      "justifyContent": "center",
      "marginTop": 16,
      "opacity": 1,
      "shadowColor": "rgba(194, 193, 248, 0.3)",
      "shadowOffset": Object {
        "height": 7,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 5,
      "width": "80%",
    }
  }
>
  <Text
    color="#FFFF"
    size="h6"
    style={
      Array [
        Object {
          "color": "#FFFF",
          "fontFamily": "WorkSans-Bold",
          "fontSize": 14,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
        },
      ]
    }
    testID="undefined_text"
  >
    teste
  </Text>
</View>
`;
